src/main.rs

```diff
@@ -269,6 +298,25 @@ impl Game {
         self.snake.parts.insert(new_head_pos);
         Move::Ok
     }
+    //returns true if new high score, and the previous high score
+    fn game_finish(self: &mut Game) -> (bool, u32) {
+        //save preferably in $HOME/.snake but fallback to /etc/.snake
+        let mut home = match home::home_dir() {
+            Some(path) => path,
+            None => PathBuf::from(r"/etc/"),
+        };
```

Comment:
I don't think `/etc/` is a good fallback to `$HOME`. I'd rather it just wouldn't save the highscore if `$HOME` is not defined.